java/src/main/java/io/cucumber/java/JavaHookDefinition.java

```diff
@@ -22,35 +27,74 @@
 
     private static Method requireValidMethod(Method method) {
         Class<?>[] parameterTypes = method.getParameterTypes();
-        if (parameterTypes.length > 1) {
-            throw createInvalidSignatureException(method);
+        Class<?>[] acceptedTypes = new Class<?>[] { io.cucumber.java.Scenario.class };
+
+        if (method.isAnnotationPresent(BeforeStep.class) || method.isAnnotationPresent(AfterStep.class)) {
+            acceptedTypes = new Class<?>[] { io.cucumber.java.Scenario.class, io.cucumber.core.gherkin.Step.class };
```

Comment:
The arguments types can be in any order and neither is required.

Reply:
Also the `io.cucumber.core.gherkin.Step` should not be exposed to the end user. The user should only have to use imports from `io.cucumber.java`.